src/App.jsx

```diff
@@ -4,32 +4,81 @@ import viteLogo from '/vite.svg'
 import './App.css'
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [isFormVisible, setIsFormVisible] = useState(false);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [message, setMessage] = useState('');
+  const [isFormValid, setIsFormValid] = useState(true);
+
+  const handleShowForm = () => {
+    setIsFormVisible(true);
+  };
+
+  const handleClearForm = () => {
+    setName('');
+    setEmail('');
+    setMessage('');
+    setIsFormValid(true);
+  };
+
+  const handleSubmit = () => {
+    if (name.trim() === '' || email.trim() === '' || message.trim() === '') {
```

Comment:
I'd recommend creating a separate function for handling each input field. This way, you only run the validation code you need. This approach would also help with adding the "live validation" requirement which was missed.

I would have liked to see different types of validation instead of only checking each field was blank.